ContactsListVer2/ContactListTVCPresenter.swift

```diff
@@ -0,0 +1,272 @@
+        cell.presenter = presenter
+        
+        presenter.fillCellByContact(constraintsConstant: constraintsConstant)
+    }
+    
+    func viewDidLoad() {
+        self.contactsListTVC.sortMethodSegmentControlSelectedSegmentIndex = currentSortField.rawValue
+        
+        readContactList()
+        
+        setAvailability()
+    }
+
+    func changeSortMethod() {
+        currentSortField = AdditionalData.SortFields.Values(rawValue: self.contactsListTVC.sortMethodSegmentControlSelectedSegmentIndex) ?? .lastName
+        
+        readContactList()
+    }
+
+    func readContactList() {
+        if let contactListValue = contactList {
+            contactListArray = contactListValue.getList(currentSortField: currentSortField)
+        }
+    }
+    
+    func setAvailability() {
+        
+        if contactListArray.count == 0 {
+            self.contactsListTVC.viewBackButtonIsEnabled = false
+            
+            self.contactsListTVC.viewIsEditing = false
+        } else {
+            self.contactsListTVC.viewBackButtonIsEnabled = true
+        }
+        
+        self.contactsListTVC.sortMethodSegmentControlIsHidden = contactList?.count ?? 0 <= 1
+    }
+
+    func setAvailabilityAndReloadData() {
+        readContactList()
+        
+        setAvailability()
+
+        self.contactsListTVC.reloadData()
+    }
+    
+    func setViewBackButtonTitle(_ title: String) {
+        self.contactsListTVC.viewBackButtonTitle = title
+    }
+    
+    func setViewIsEditing(_ value: Bool) {
+        self.contactsListTVC.viewIsEditing = value
+    }
+    
+    func tableViewdidSelectRow() {
+        if let contact = selectedContact,
+            let cellPresenter = cellsPresentersDictionary[contact.uuid] {
+            
+            if cellPresenter.showDetailInfo{
+                self.router.showViewContact()
+            } else {
+                cellPresenter.showDetailInfo = true
+                
+                cellPresenter.contactsTVCell.reloadDataFromPresenter()
+            }
+        }
+    }
+    
+    func getContactAddAlert() -> ContactAddAlert {
+        let contactAddAlert = ContactAddAlert()
+        
+        let presenter = ContactAddAlertPresenter(contactAddAlert: contactAddAlert, contactList: self.contactList)
+        
+        contactAddAlert.presenter = presenter
+        
+        return contactAddAlert
+    }
+}
+
+// tableView functions
+extension ContactListTVCPresenter {
+    func deleteContactForRow(_ row: Int) {
+        selectedContact = contactListArray[row]
+        
+        self.contactsListTVC.presentDeletionAlert(contactFullName: selectedContact?.fullName ?? "Contact") {
+            self.contactList?.deleteContact(self.selectedContact!)
+        }
+    }
+    
+    func selectContactForRow(_ row: Int) {
+        selectedContact = contactListArray[row]
+    }
+}
+
+// observers functions
+extension ContactListTVCPresenter {
+    @objc func updateContact(_ notification: NSNotification) {
```

Comment:
Suggest usage of Notification/NotificationCenter instead of NSNotification/NSNotificationCenter